lib/screens/sign/transaction_details_widget.dart

```diff
@@ -32,207 +33,224 @@ class TransactionDetailsWidget extends StatelessWidget {
 
     return SizedBox(
       width: MediaQuery.of(context).size.width,
-      child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
-        Container(
-          padding: const EdgeInsets.only(left: defaultPadding * 1.5, right: defaultPadding * 1.5, top: defaultPadding * 3),
-          child: Text(
-            'Approve transaction?',
-            style: textTheme.displayMedium?.copyWith(fontWeight: FontWeight.bold),
-          ),
-        ),
-        const Gap(defaultPadding),
-        Container(
-          padding: const EdgeInsets.all(defaultPadding * 1.5),
-          child: Row(
-            children: [
-              Consumer<KeysharesProvider>(builder: (context, keysharesProvider, _) {
-                var address = keysharesProvider.keyshares.firstOrNull?.ethAddress;
-                return WalletAddressWidget(
-                  title: 'From wallet',
-                  // displayText: address != null ? "${address.substring(0, 5)}....${address.substring(address.length - 5)}" : "Unknown address",
-                  displayText: address ?? "Unknown address",
-                  copyText: address!,
-                  crossAxisAlignment: CrossAxisAlignment.start,
-                );
-              }),
-              const Spacer(),
-              if (requestModel.recipient != null)
-                WalletAddressWidget(
-                  title: 'To wallet',
-                  displayText: requestModel.recipient!,
-                  // '${requestModel.recipient!.substring(0, 5)}...${requestModel.recipient!.substring(requestModel.recipient!.length - 5)}',
-                  copyText: requestModel.recipient!,
-                  crossAxisAlignment: CrossAxisAlignment.end,
-                ),
-            ],
-          ),
-        ),
-        Container(
-          padding: const EdgeInsets.all(defaultPadding * 1.5),
-          width: MediaQuery.of(context).size.width,
-          color: backgroundPrimaryColor.withOpacity(0.1),
-          child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
-            if (requestModel.signType.isTransaction) ...[
-              Text(
-                "Message",
-                style: textTheme.bodyMedium,
+      child: FutureBuilder<Chain>(
+        future: requestModel.chain,
+        builder: (context, snapshot) {
+          if (snapshot.connectionState == ConnectionState.waiting) {
+            return Center(
+              child: Lottie.asset('assets/lottie/MobileLoader.json'),
+            );
+          }
+          if (snapshot.hasError) {
```

Comment:
If we failed to load chain info - it probably doesn't mean that we should abort transaction signing. As its not the crucial info - we can still show to user UI without details about chain (besides chain id).